### PR TITLE
[MRG] hyperband save and reload

### PR DESCRIPTION
**Pull request by jhfjhfj1**

### CODE REVIEW

File: kerastuner/tuners/hyperband.py [hunk 1]
```diff
@@ -195,14 +203,59 @@ def _get_epoch_sequence(self):
...
-        # TODO
-        raise NotImplementedError
+    def save(self, fname):
+        state = {
+            'seed': self.seed,
+            'factor': self.factor,
+            'min_epochs': self.min_epochs,
+            'max_epochs': self.max_epochs,
+            'queue': queue_to_list(copy.copy(self._queue)),
+            'trial_count': self._trials_count,
```
Comment (fchollet) [inline kerastuner/tuners/hyperband.py]: Inconsistency: `trial_count` vs `trials_count`, rename to `trial_count`

File: tests/kerastuner/tuners/hyperband_test.py [hunk 2]
```diff
@@ -93,6 +94,32 @@ def test_hyperband_dynamic_space(tmp_dir):
...
+
+    fname = os.path.join(tmp_dir, 'oracle')
+    oracle.save(fname)
+    oracle = hyperband_module.HyperbandOracle()
+    oracle.reload(fname)
+
+    for trial_id in range(oracle._model_sequence[0] - 2):
+        hp = oracle.populate_space('1_' + str(trial_id), hp_list)
+        assert hp['status'] == 'RUN'
+    assert oracle.populate_space('idle', hp_list)['status'] == 'IDLE'
```
Comment (fchollet) [inline tests/kerastuner/tuners/hyperband_test.py]: Are these checks sufficient to assert that the oracle is correctly loaded and ready to resume search?

Comment (jhfjhfj1) [reply tests/kerastuner/tuners/hyperband_test.py]: I have added more corner cases tests of when the oracle is stopped and reloaded. I think it should be enough.
